neo4j/result.go

```diff
@@ -23,8 +23,9 @@ import (
 	"context"
 )
 
+// Result is created via the context-aware Session type.
```

Comment:
Is it? I thought, at least as it stands right now, it's created via either a session (with or without context) or a transaction (with or without context).

Reply:
yeah, the comment is bogus
`Result` is created via `Session`, which is **not** "context-aware.
`ResultWithContext` is created via `SessionWithContext`, which is "context-aware".

Reply:
```suggestion
// Result is created via the Session type.
```